llama-index-integrations/readers/llama-index-readers-oracleai/llama_index/readers/oracleai/base.py

```diff
@@ -0,0 +1,457 @@
+                        if row is None:
+                            doc_id = OracleDocReader.generate_object_id(
+                                self.conn.username
+                                + "$"
+                                + self.owner
+                                + "$"
+                                + self.tablename
+                                + "$"
+                                + self.colname
+                            )
+                            metadata["_oid"] = doc_id
+                            results.append(Document(text="", metadata=metadata))
+                        else:
+                            if row[0] is not None:
+                                data = str(row[0])
+                                if data.startswith(("<!DOCTYPE html", "<HTML>")):
+                                    p = ParseOracleDocMetadata()
+                                    p.feed(data)
+                                    metadata = p.get_metadata()
+
+                            doc_id = OracleDocReader.generate_object_id(
+                                self.conn.username
+                                + "$"
+                                + self.owner
+                                + "$"
+                                + self.tablename
+                                + "$"
+                                + self.colname
+                                + "$"
+                                + str(row[2])
+                            )
+                            metadata["_oid"] = doc_id
+                            metadata["_rowid"] = row[2]
+
+                            # process projected metadata cols
+                            if self.mdata_cols is not None:
+                                ncols = len(self.mdata_cols)
+
+                            for i in range(ncols):
+                                if i == 0:
+                                    metadata["_rowid"] = row[i + 2]
+                                else:
+                                    metadata[self.mdata_cols[i]] = row[i + 2]
+
+                            if row[1] is None:
+                                results.append(Document(text="", metadata=metadata))
+                            else:
+                                results.append(
+                                    Document(text=str(row[1]), metadata=metadata)
+                                )
+                except Exception as ex:
+                    print(f"An exception occurred :: {ex}")
+                    traceback.print_exc()
+                    cursor.close()
+                    raise
+
+            return results
+        except Exception as ex:
+            print(f"An exception occurred :: {ex}")
+            traceback.print_exc()
+            raise
+
+
+import logging
+from typing import (
+    Any,
+    Dict,
+    List,
+)
+
+
+logger = logging.getLogger(__name__)
+
+
+class OracleTextSplitter:
+    """Splitting text using Oracle chunker."""
```

Comment:
Curious why the text splitter is in the reader class?

Reply:
Usually, after reading or loading the docs, users want to split the docs. So we put it in the same file. Also, we did not find any textsplit class or category we were looking for in the project.

Reply:
thanks

Reply:
@forkhroy we have a `NodeParser` class that we generally use for text splitting and other related parsing techniques.

If the splitting only works on the data coming from the reader though, maybe we should just bundle them together, have the reader have an option to output split text? Not sure tbh

Reply:
yeah, we just bundled them together just like we did the same for others.

Reply:
@forkhroy I kind of meant including an option in the reader to split the text, rather than a separate class... my point is, users expect certain imports to be in certain places, and they also expect modules to follow existing interfaces. 

If we ignore this, we end up with a pile of code rather than an actual framework.